node/pkg/driver/node_test.go

```diff
@@ -57,73 +96,350 @@ func TestNodeStageVolume(t *testing.T) {
 			Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
 		},
 	}
+	publishContext := map[string]string{PublishContextParamLun: "1", PublishContextParamConnectivity: "iSCSI"}
+	stagingRequest := &csi.NodeStageVolumeRequest{
+		PublishContext:    publishContext,
+		StagingTargetPath: stagingPath,
+		VolumeCapability:  stdVolCap,
+		VolumeId:          volId,
+	}
+
 	testCases := []struct {
-		name       string
-		req        *csi.NodeStageVolumeRequest
-		expErrCode codes.Code
+		name     string
+		testFunc func(t *testing.T)
```

Comment:
if test funcs are different, what is the reason to add them to the same test cases suite?

Reply:
They check the same API method in node,go

Reply:
This is how I saw i tutorial and also the same way did Amazon csi driver:
https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/9ca95cea4aaba3b03ae5346520070e629509f748/pkg/driver/node_test.go